Sources/Server/ServerNetworkLayer.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by Rafał Rzeuski on 09.05.16.
+//
+
+#include "ServerNetworkLayer.h"
+#include "../Libraries/json/json.h"
+#include <iostream>
+#include <string>
+
+int ServerNetworkLayer::connectToNode(const std::string addressWithPort) {
+    nodeConnectionSocket = new TCPSocket();
+    SocketAddress nodeAddress(addressWithPort);
+    // No local port binding needed, just connect(...)
+    if(nodeConnectionSocket->connect(nodeConnectionSocket->internalDescriptor(), nodeAddress) == -1) {
+        std::cerr << "Failed to connect to target address!";
+        return -1;
+    }
+    return 0;
```

Comment:
We need to hold info about layer being connected somewhere(or maybe check it dynamically). It's a nice to have thing.